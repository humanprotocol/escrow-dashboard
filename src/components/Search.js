import { useState } from 'react';
import Web3 from 'web3';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import InputAdornment from '@mui/material/InputAdornment';
import SearchIcon from '@mui/icons-material/Search';
import IconButton from '@mui/material/IconButton';

export default function Search({onSetEscrow}) {
  const [escrow, setEscrow] = useState('');
  const isEscrowValid = Boolean(escrow) && Web3.utils.isAddress(escrow);
  const keyDownHandle = (e) => {
    if (e.code == "Enter") {
      if (!escrow && !isEscrowValid ) return;
      onSetEscrow(escrow);
    }
  };
  return (
    <Box sx={{width: "calc(100%)", display: "flex", flexDirection: "row", flexGrow: 4, borderRadius: "3px"}}>
      <TextField
        variant="outlined"
        size="small"
        label="Escrow Factory Address"
        sx={{width: "calc(100%)",  flexBasis: "100%"}}
        onChange={(e) => setEscrow(e.target.value)}
        onKeyDown={keyDownHandle}
        > 
          {escrow}
        </TextField>
        <IconButton type="submit" sx={{backgroundColor:'#320A8D', borderRadius:'0 3px 3px 0', color:'white', marginLeft:'-2px', cursor: 'pointer'}} aria-label="search" onClick={ keyDownHandle }>
          <SearchIcon />
        </IconButton>
    </Box>
  )
}
