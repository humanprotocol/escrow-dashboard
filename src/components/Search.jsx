import React, { useState } from 'react';
import Web3 from 'web3';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

export default function Search({ onSetEscrow }) {
  const [escrow, setEscrow] = useState('');
  const isEscrowValid = Boolean(escrow) && Web3.utils.isAddress(escrow);

  return (
    <Box sx={{ display: 'flex', flexDirection: 'row', mt: 3, flexGrow: 4 }}>
      <TextField
        variant="outlined"
        size="small"
        label="Escrow Factory Address"
        sx={{ flexBasis: '90%', mr: 1 }}
        onChange={(e) => setEscrow(e.target.value)}
      >
        {escrow}
      </TextField>
      <Button
        variant="outlined"
        onClick={() => onSetEscrow(escrow)}
        disabled={!isEscrowValid}
      >
        Search
      </Button>
    </Box>
  );
}
