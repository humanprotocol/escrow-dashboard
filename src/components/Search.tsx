import * as React from 'react';
import Web3 from 'web3';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import { INDENT } from 'src/ui';

interface ISearch {
  onSetEscrow: (value: string) => void;
}

const Search: React.FC<ISearch> = ({ onSetEscrow }): React.ReactElement => {
  const [escrow, setEscrow] = React.useState('');
  const isEscrowValid = Boolean(escrow) && Web3.utils.isAddress(escrow);

  return (
    <Box
      sx={{ display: 'flex', flexDirection: 'row', mt: INDENT.L, flexGrow: 4 }}
    >
      <TextField
        variant="outlined"
        size="small"
        label="Escrow Factory Address"
        sx={{ flexBasis: '90%', mr: 1 }}
        onChange={(e) => setEscrow(e.target.value)}
      >
        {escrow}
      </TextField>
      <Button
        variant="outlined"
        onClick={() => onSetEscrow(escrow)}
        disabled={!isEscrowValid}
      >
        Search
      </Button>
    </Box>
  );
};

export default Search;
