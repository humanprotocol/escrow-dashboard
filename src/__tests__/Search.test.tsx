import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Search from 'src/components/Search';

const props = {
  onSetEscrow: () => {},
};

describe('when rendered with a `name` prop', () => {
  it('should paste it into the greetings text', () => {
    render(<Search onSetEscrow={props.onSetEscrow} />);
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Search onSetEscrow={props.onSetEscrow} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
