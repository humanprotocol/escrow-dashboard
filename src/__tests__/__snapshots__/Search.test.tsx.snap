// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-uq2i2m"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1jj4e84-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
    >
      Escrow Factory Address
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            Escrow Factory Address
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    Search
  </button>
</div>
`;
